src/main/java/com/khutircraftubackend/category/exception/category/CategoryExceptionMessages.java

```diff
@@ -7,4 +7,5 @@ public class CategoryExceptionMessages{
     public static final String CATEGORY_NOT_FOUND = "Category not found";
     public static final String CATEGORY_HAS_SUBCATEGORIES_OR_PRODUCTS =
             "Category has subcategories or products, can't be deleted";
+    public static final String CATEGORY_HAS_PRODUCTS = "Category has products";
```

Comment:
this constant "CATEGORY_HAS_PRODUCTS" not required any more, coz "CATEGORY_HAS_SUBCATEGORIES_OR_PRODUCTS" covers existence of the underlying products in a category.